File: C++/4891.cpp
#include <iostream>
#include <stdlib.h>

using namespace std;

int gapScores(int* pscores, int len) {
	int i;
	int min = *(pscores);
	int max = *(pscores);
	
	for(i = 1; i < len; i++)
	{
		if( *(pscores+i) < min )
			min = *(pscores+i);
		if( *(pscores+i) > max )
			max = *(pscores+i);
	}
	
	return max - min;
}

int main(void)
{
	// 입력 및 초기화 
	int N, i;
	cin >> N;
	int* scores = (int*)malloc(sizeof(int)*N);
	
	for(i = 0; i < N; i++)
		cin >> *(scores + i);
	
	// 출력부 
	cout << gapScores(scores, N);
}

File: Java/4891.java
import java.util.Arrays;
import java.util.OptionalInt;
import java.util.Scanner;

public class Main {
    /* 메인 함수 */
    public static void main(String[] args)  {
        Scanner scanner = new Scanner(System.in);

        // 입력 및 초기화
        int N = scanner.nextInt();
        int[] scores = new int[N];

        for(int i = 0; i < N; i++)
            scores[i] = scanner.nextInt();

        // 계산부
        OptionalInt max = Arrays.stream(scores).max();
        OptionalInt min = Arrays.stream(scores).min();

        int maxInt = max.orElse(0);
        int minInt = min.orElse(0);

        // 출력부
        System.out.println(maxInt - minInt);
    }
}